nixos/modules/services/matrix/mautrix-meta.nix

```diff
@@ -0,0 +1,210 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-meta;
+  fullDataDir = "/var/lib/${cfg.dataDir}";
+  relativeDataDir = cfg.dataDir;
+  pkg = cfg.package;
+  registrationFile = "${fullDataDir}/meta-registration.yaml";
```

Comment:
Oh. Since I use conduit, I have not realized there is the problem with permissions, thats a good point. I should test it against that. 

I agree its sort of a workaround for managing secrets. I like it allows either opting into nixops/similar via the config, or via env file where you can avoid that setup, while still keeping secrets not readable to the world. The other approach basically means necessity of using something like nixops, as far as I understand it. 

I am still open to it though.

Reply:
Yeah, I guess with standard NixOS the other approach isn't very easy as there is no way to get data out other than the Nix store. Maybe this generated approach is best for now. However we do need some solution to the permission issue. Maybe just explicitly changing the permissions on the file is fine for now. Or creating a separate directory to contain this file and make it owned by a group that has both the matrix server and the bridge? Maybe it would also be best to split this into a separate systemd unit. That way both the bridge and homeserver can depend on the registration generation, then the homeserver can start, then the bridge proper can depend on the homeserver.

Reply:
Could you please provide an example on how to put in as_token, hs_token without putting them to the nix store when mautrix-facebook approach is used? I've been playing with this more, and feeling like this registration generation is not really the way to go and that mautrix-facebook approach could be better.

Reply:
Let's continue discussing this here: https://github.com/NixOS/nixpkgs/pull/296718#discussion_r1545729335